src/app/modules/main/communities/module.nim

```diff
@@ -513,4 +514,45 @@ method prepareTokenModelForCommunity*(self: Module, communityId: string) =
     let tokenPermissionItem = buildTokenPermissionItem(tokenPermission, chats)
     tokenPermissionsItems.add(tokenPermissionItem)
 
-  self.view.spectatedCommunityPermissionModel.setItems(tokenPermissionsItems)
\ No newline at end of file
+  self.view.spectatedCommunityPermissionModel.setItems(tokenPermissionsItems)
+
+method updateTokenModelForCommunity*(self: Module, communityId: string, sharedAddresses: seq[string]) =
+  self.controller.asyncCheckPermissionsToJoin(communityId, sharedAddresses)
+
+method onCommunityCheckPermissionsToJoinResponse*(self: Module, communityId: string, sharedAddresses: seq[string], checkPermissionsToJoinResponse: CheckPermissionsToJoinResponseDto) =
```

Comment:
I don't think you need `sharedAddresses`. It should be passed to the service to do the checks and here you don,t use them (you don't need to keep them as tmp in that case)